### PR TITLE
fix: [CDS-78028]: added ResourceHandlerTable as common component

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/microfrontends/index.ts [hunk 1]
```diff
@@ -137,5 +140,7 @@ export {
   GovernanceCustomMicroFrontendProps,
   ETCustomMicroFrontendProps,
   SRMCustomMicroFrontendProps,
-  SEICustomMicroFrontendProps
+  SEICustomMicroFrontendProps,
+  RbacResourceModalProps,
+  RbacResourceRendererProps
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/index.ts:145]: these should not be needed to be exported

File: src/microfrontends/index.ts [hunk 2]
```diff
@@ -72,6 +74,7 @@ export interface CommonComponents {
   LevelUpBanner?: typeof LevelUpBanner
   ParentLink?: typeof ParentLink
   SideNav?: typeof SideNav
+  ResourceHandlerTable: typeof ResourceHandlerTable
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/index.ts:77]: this should be optional

File: src/microfrontends/ChildAppMounter.tsx [hunk 3]
```diff
@@ -49,7 +51,8 @@ function ChildAppMounter<T>({ ChildApp, children, ...rest }: T & ChildAppMounter
         MonacoDiffEditor,
         LevelUpBanner,
         ParentLink,
-        SideNav
+        SideNav,
+        ResourceHandlerTable
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/ChildAppMounter.tsx:55]: How are other microfrontends handling it today? Is it passed in custom props for each MFE?
it might be better to do the same for PM mfe as well, just to avoid side-effects on other MFEs.

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/microfrontends/ChildAppMounter.tsx (hunk 3 above)
Comment (sahithikolichala) [reply src/microfrontends/ChildAppMounter.tsx:55]: Other MFES have it implemented within harness-core-ui ...

File: src/microfrontends/index.ts (hunk 1 above)
Comment (sahithikolichala) [reply src/microfrontends/index.ts:145]: but we need this for resource listing

File: src/microfrontends/index.ts [hunk 4]
```diff
@@ -72,6 +73,7 @@ export interface CommonComponents {
   LevelUpBanner?: typeof LevelUpBanner
   ParentLink?: typeof ParentLink
   SideNav?: typeof SideNav
+  ResourceHandlerTable?: typeof ResourceHandlerTable
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/index.ts:76]: We don't need this anymore, right?

File: src/microfrontends/ChildAppMounter.tsx [hunk 5]
```diff
@@ -25,6 +25,7 @@ import LevelUpBanner from '@common/components/FeatureWarning/LevelUpBanner'
 import ParentLink from '@common/components/ParentLink/ParentLink'
 import { getLocationPathName } from 'framework/utils/WindowLocation'
 import { useEventSourceListener } from '@common/hooks/useEventSourceListener'
+
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/ChildAppMounter.tsx:28]: unnecessary change in this file

File: src/modules/25-governance/GovernanceApp.tsx [hunk 6]
```diff
@@ -9,6 +9,7 @@ import React, { lazy, FC } from 'react'
 import { useRouteMatch } from 'react-router-dom'
 import ChildAppMounter from 'microfrontends/ChildAppMounter'
 import routes from '@common/RouteDefinitions'
+
```
Comment (abhinavrastogi-harness) [inline src/modules/25-governance/GovernanceApp.tsx:12]: unnecessary changes in this file

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Review by tan-nhu — APPROVED** [review]
